desktop/sources/scripts/cursor.js

```diff
@@ -108,11 +128,15 @@ export default function Cursor (terminal) {
     terminal.history.record(terminal.orca.s)
   }
 
-  this.erase = function () {
-    this.eraseBlock(this.x, this.y, this.w, this.h)
-    if (this.mode === 1) { this.move(-1, 0) }
-    terminal.history.record(terminal.orca.s)
-  }
+  this.erase = function() {
+    for (let y = this.minY; y <= this.maxY; y++) {
+      for (let x = this.minX; x <= this.maxX; x++) {
+        terminal.orca.write(x, y, ".");
+      }
+    }
+
+    terminal.history.record(terminal.orca.s);
```

Comment:
@neauoire I've made the changes you mentioned! This is the only remaining part of the code that I'm unsure about. Is there something that was happening here before that isn't now? I wasn't sure what the `if(this.mode === 1)` meant. Maybe it would be good to throw some constants in for the mode values so it's clearer what they mean for people modifying the code?

Reply:
The `mode 1` is insert, could you double-check and make sure that nothing breaks if you're in insert-mode `cmd+i`? I can't test atm, I will be back home soon.

Reply:
I made one small change to set the width and height properly when switching modes, but otherwise it seems fine to me!

Reply:
Awesome, yeah it looks good here as well!